client/src/main/scala/org/apache/celeborn/client/commit/CommitHandler.scala

```diff
@@ -357,6 +357,8 @@ abstract class CommitHandler(
             logError(
               s"Ask worker($worker) CommitFiles for $shuffleId failed because of Timeout" +
                 s" (attempt ${status.retriedTimes}/$maxRetries), will not retry.")
+            val res = createFailResponse(status)
+            processResponse(res, status.workerInfo)
             iter.remove()
```

Comment:
I think it would be better to remove this line. And let the final checks to handle all timeout requests.

Reply:
Where is the processing logic of final check? After iter.remove() is called, the status information has been removed from the collection